tensorflow_recommenders_addons/dynamic_embedding/python/kernel_tests/dynamic_embedding_variable_test.py

```diff
@@ -1051,6 +1095,138 @@ def test_dynamic_embedding_variable_with_random_init(self):
 
+  def test_dynamic_embedding_variable_with_restrict_v1(self):
+    if context.executing_eagerly():
+      self.skipTest('skip eager test when using legacy optimizers.')
+
+    optmz = de.DynamicEmbeddingOptimizer(adam.AdamOptimizer(0.1))
+    data_len = 32
+    maxval = 256
+    residue = 100
+    trigger = 150
+    embed_dim = 8
+
+    var_guard_by_tstp = de.get_variable(
+        'tstp_guard',
+        key_dtype=dtypes.int64,
+        value_dtype=dtypes.float32,
+        initializer=-1.,
+        dim=embed_dim,
+        restrict_policy=de.TimestampRestrictPolicy)
+
+    var_guard_by_freq = de.get_variable(
+        'freq_guard',
+        key_dtype=dtypes.int64,
+        value_dtype=dtypes.float32,
+        initializer=-1.,
+        dim=embed_dim,
+        restrict_policy=de.FrequencyRestrictPolicy)
+
+    sparse_vars = [var_guard_by_tstp, var_guard_by_freq]
+
+    indices = [data_fn((data_len, 1), maxval) for _ in range(3)]
+    _, trainables, loss = model_fn(sparse_vars, embed_dim, indices)
+    train_op = optmz.minimize(loss, var_list=trainables)
+
+    var_sizes = [0, 0]
+    self.evaluate(variables.global_variables_initializer())
+
+    while not all(sz > trigger for sz in var_sizes):
+      self.evaluate(train_op)
+      var_sizes = self.evaluate([spv.size() for spv in sparse_vars])
+
+    self.assertTrue(all(sz >= trigger for sz in var_sizes))
+    tstp_restrict_op = var_guard_by_tstp.restrict(residue, trigger=trigger)
+    if tstp_restrict_op != None:
+      self.evaluate(tstp_restrict_op)
+    freq_restrict_op = var_guard_by_freq.restrict(residue, trigger=trigger)
+    if freq_restrict_op != None:
+      self.evaluate(freq_restrict_op)
+    var_sizes = self.evaluate([spv.size() for spv in sparse_vars])
+    self.assertAllEqual(var_sizes, [residue, residue])
+
+    slot_params = []
+    for _trainable in trainables:
+      slot_params += [
+          optmz.get_slot(_trainable, name).params
+          for name in optmz.get_slot_names()
+      ]
+    slot_params = list(set(slot_params))
+
+    for sp in slot_params:
+      self.assertAllEqual(self.evaluate(sp.size()), residue)
+    tstp_size = self.evaluate(var_guard_by_tstp.restrict_policy.status.size())
+    self.assertAllEqual(tstp_size, residue)
+    freq_size = self.evaluate(var_guard_by_freq.restrict_policy.status.size())
+    self.assertAllEqual(freq_size, residue)
+
+  def test_dynamic_embedding_variable_with_restrict_v2(self):
+    if not context.executing_eagerly():
+      self.skipTest('Test in eager mode only.')
+
+    optmz = de.DynamicEmbeddingOptimizer(optimizer_v2.adam.Adam(0.1))
+    data_len = 32
+    maxval = 256
+    residue = 100
+    trigger = 150
+    embed_dim = 8
+    trainables = []
+
+    var_guard_by_tstp = de.get_variable(
+        'tstp_guard',
+        key_dtype=dtypes.int64,
+        value_dtype=dtypes.float32,
+        initializer=-1.,
+        dim=embed_dim,
+        restrict_policy=de.TimestampRestrictPolicy)
```

Comment:
setting a small `init_size` for all get_variable maybe better for weak resource on github CI env.

Reply:
Agree. 256 is used.